scripts/workloadidentityfederation-setup.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+suffix=""
+
+function generate_random_string () {
+  local valid_chars=abcdefghijklmnopqrstuvwxyz0123456789
+  for i in {1..8} ; do
+    suffix+="${valid_chars:RANDOM%${#valid_chars}:1}"
+    done
+}
+
+generate_random_string
+
+pool_id="pool-"$suffix
+oidc_provider_id="oidc-"$suffix
+aws_provider_id="aws-"$suffix
+
+# Fill in.
+project_id=""
+project_number=""
+aws_account_id=""
+aws_role_name=""
+service_account_email=""
+sub=""; # client_id from service account key file
+
+oidc_aud="//iam.googleapis.com/projects/$project_number/locations/global/workloadIdentityPools/$pool_id/providers/$oidc_provider_id"
+aws_aud="//iam.googleapis.com/projects/$project_number/locations/global/workloadIdentityPools/$pool_id/providers/$aws_provider_id"
+
+gcloud config set project $project_id
+
+# Create the Workload Identity Pool.
+gcloud beta iam workload-identity-pools create $pool_id \
```

Comment:
variable injection should always use `"${var_name}"` to be safe, especially if using the variables in bash arguments

```suggestion
gcloud beta iam workload-identity-pools create "${pool_id}" \
```